fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+
+    fetch("http://localhost:3000/api/todos")
+      .then(res=>res.json())
+      .then(data=>this.setState({todos:data}))
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+  }
+
+  addTodo(newTodo) {
+
+    fetch("http://localhost:3000/api/todos", {
+      method:"POST",
+      headers:{"Content-Type": "application/json"},
+      body: JSON.stringify(newTodo)
+    })
+      .then(res=>res.json())
+      .then(newTodo=>{
+        this.state.todos.push(newTodo)
```

Comment:
You don't want to push the new todo into the array, you want to make a copy of the array and add the todo to the end of the new array. Check out this article on the best ways to do that: https://www.samanthaming.com/tidbits/35-es6-way-to-clone-an-array/